File: ticket-app/src/models/ticket.ts
import { Project, User, WorkGroup } from "@models";
import { TicketPriorityLevel, TicketStatus } from "@viewModels";
import { BaseModel } from "./baseModel";

export class Ticket extends BaseModel {
    title: string;
    description: string;
    priority: TicketPriorityLevel;
    projectId: number;
    project: Project;
    status: TicketStatus;
    assignedToId: number | null;
    assignedTo: WorkGroup | null;
    responsibleId: number | null;
    responsible: User | null;
    flags: number[];
}

File: ticket-api/Entities/Ticket.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ticket_api.Entities;

public class Ticket : BaseEntityWithUser
{
    public string Title { get; set; }
    public string Description { get; set; }
    public TicketPriorityLevel Priority { get; set; }
    public long ProjectId { get; set; }
    public Project Project { get; set; }
    public TicketStatus Status { get; set; }
    public long? AssignedToId { get; set; }
    public WorkGroup? AssignedTo { get; set; }
    [Column(TypeName = "Date")]
    public DateTime? StartDate { get; set; }
    [Column(TypeName = "Date")]
    public DateTime? EndDate { get; set; }
    [Column(TypeName = "Date")]
    public DateTime? ActualFinishingDate { get; set; }
    public long? ResponsibleId { get; set; }
    public AppUser? Responsible { get; set; }
    public List<long>? Flags { get; set; } = new List<long>();
}
